lib/MongoDB/Role/_RetryableBulk.pm

```diff
@@ -0,0 +1,39 @@
+#
+#  Copyright 2016 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+package MongoDB::Role::_RetryableBulk;
```

Comment:
Having a role just to provide an attribute seems excessive.  I think it's OK to just put it in the two classes that need it.